packages/p2p/src/components/my-ads/create-ad-form.jsx

```diff
@@ -365,6 +369,18 @@ const CreateAdForm = () => {
                 }}
             </Formik>
             <CreateAdErrorModal />
+            <CancelAddPaymentMethodModal
+                onCancel={() => {
+                    my_ads_store.setShouldShowAddPaymentMethod(false);
+                }}
+                onGoBack={() => {
+                    if (isMobile()) {
+                        my_ads_store.setShouldShowAddPaymentMethodModal(true);
+                    } else {
+                        setTimeout(() => my_ads_store.setShouldShowAddPaymentMethodModal(true), 250);
```

Comment:
what is the setTimeout for here? do we need it?

Reply:
The setTimeout is used to delay hiding the previous modal before the next modal is shown, or else there will be a bug where the previous modal will snap to the left side while the new modal is appearing, before the previous modal is hidden due to the animation